### PR TITLE
Fix unused parameter warnings

### PR DESCRIPTION
**Pull request by veblush**
- Removing `-Wno-unused-parameter`.
- Fixing all lines causing `unused parameter`
  - C++: commenting out parameters
  - C: using `(void)arg`

---
This change is [Reviewable](https://reviewable.io/reviews/grpc/grpc/25134)

### CODE REVIEW

File: src/core/lib/transport/metadata.cc [hunk 1]
```diff
@@ -673,6 +673,10 @@ void grpc_mdelem_do_unref(grpc_mdelem gmd DEBUG_ARGS) {
 
 void grpc_mdelem_on_final_unref(grpc_mdelem_data_storage storage, void* ptr,
                                 uint32_t hash DEBUG_ARGS) {
+#ifndef NDEBUG
+  (void)file;
+  (void)line;
+#endif
```
Comment (drfloob) [inline src/core/lib/transport/metadata.cc:679]: Why is this needed here? This function doesn't reference file or line variables anywhere.

File: src/cpp/ext/filters/census/context.cc [hunk 2]
```diff
@@ -28,8 +28,9 @@ using ::opencensus::tags::TagMap;
 using ::opencensus::trace::Span;
 using ::opencensus::trace::SpanContext;
 
-void GenerateServerContext(absl::string_view tracing, absl::string_view stats,
-                           absl::string_view primary_role,
+void GenerateServerContext(absl::string_view tracing,
+                           absl::string_view /*stats*/,
+                           absl::string_view /*primary_role*/,
```
Comment (drfloob) [inline src/cpp/ext/filters/census/context.cc:33]: Can these arguments be deleted?

**Review by drfloob — COMMENTED** [review]
Very cool. I'm curious, what is your tolerance level for going through and seeing if any of these unused parameters can be removed? I found maybe one spot where that'd be possible, commented on below. Many of the unused arguments are in subclass implementations, where the arguments are still meaningful for other sublcasses.

**Review by jtattermusch — APPROVED** [review]
LGTM for the general idea of removing "-Wno-unused-parameter" and to grant owner's approval.
Please still do get the C/C++ changes reviewed by someone on C core team.

**Review by markdroth — CHANGES_REQUESTED** [review]
Thanks for doing this!

Please let me know if you have any questions.  Thanks!

> Reviewed 55 of 55 files at r1.
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/25134#-:-MR0wiDE5yoacGLphw5Q:b-pevjuw)* status: all files reviewed, 6 unresolved discussions (waiting on @drfloob, @gnossen, @nicolasnoble, and @veblush)

___
*[src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_channel_secure.cc, line 43 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0sxWm0I2bDCq7P7K7:-MR0sxWn5hN1bj4rLwbK:b-8ur4cy) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_channel_secure.cc#L43)):*
> ```cpp
> 
> grpc_channel_args* ModifyGrpclbBalancerChannelArgs(
>     const ServerAddressList& /*addresses*/, grpc_channel_args* args) {
> ```

It looks like neither implementation (secure and insecure) of `ModifyGrpclbBalancerChannelArgs()` actually uses this argument.  Let's just remove it from the API and stop passing it in.
___
*[src/core/lib/surface/server.cc, line 516 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0u-u34aF8j5xwBe0K:-MR0u-u34aF8j5xwBe0L:bsp0rcn) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/src/core/lib/surface/server.cc#L516)):*
> ```cpp
> 
> RefCountedPtr<channelz::ServerNode> CreateChannelzNode(
>     Server* /*server*/, const grpc_channel_args* args) {
> ```

If this parameter isn't needed, let's just remove it from the API and change callers not to pass it.
___
*[src/core/lib/transport/metadata.cc, line 679 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0siCwQSOFz3eRx3Ti-r1-679:-MR0uA9j3HRrBGlXDart:bp5ucn6) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/src/core/lib/transport/metadata.cc#L679)):*
<details><summary><i>Previously, drfloob (Adam J Heller) wrote…</i></summary><blockquote>

Why is this needed here? This function doesn't reference file or line variables anywhere.
</blockquote></details>

The file and line variables are passed via the `DEBUG_ARGS` macro, which are defined only in debug builds.

We use this kind of macro-based hack in a number of places.  They should ideally all be rewritten to use `grpc_core::DebugLocation` instead, but that's probably outside of the scope of this PR.
___
*[src/cpp/ext/filters/census/context.cc, line 33 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0siCwQSOFz3eRx3Tm-r1-33:-MR0uolp05dKX2zbapUC:b-p5od3x) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/src/cpp/ext/filters/census/context.cc#L33)):*
<details><summary><i>Previously, drfloob (Adam J Heller) wrote…</i></summary><blockquote>

Can these ~arguments~ parameters be deleted?
</blockquote></details>

+1.  Doesn't seem very useful to have them here, since this isn't implementing some existing API.
___
*[test/core/end2end/cq_verifier.cc, line 307 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0vWzz834oDrFIjFFh:-MR0vWzz834oDrFIjFFi:b-mrpaow) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/test/core/end2end/cq_verifier.cc#L307)):*
> ```cpp
> 
> void cq_maybe_expect_completion(cq_verifier* v, const char* file, int line,
>                                 void* tag, bool /*success*/, bool* seen) {
> ```

Hmm... Is this one a bug?  Are we sure that we shouldn't actually be using `success` here instead of hard-coding it to true?
___
*[test/cpp/end2end/xds_end2end_test.cc, line 1060 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0wYDu-ZQR8KKD8Ynv:-MR0wYDu-ZQR8KKD8Ynw:b-ffevfr) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/test/cpp/end2end/xds_end2end_test.cc#L1060)):*
> ```cpp
>       const ResourceTypeState& resource_type_state,
>       const ResourceState& resource_state, int client_resource_type_version,
>       SubscriptionState* /*subscription_state*/) {
> ```

If this isn't actually being used, let's just remove the parameter.

**Review by veblush — COMMENTED** [review]
Thank you for the thorough review

> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/25134#-:-MRSZjBR80uTbcdt6rzR:b7gtmjc)* status: all files reviewed, 6 unresolved discussions (waiting on @drfloob, @gnossen, @markdroth, @nicolasnoble, and @yashykt)

___
*[src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_channel_secure.cc, line 43 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0sxWm0I2bDCq7P7K7:-MRSUg-KE7EhZXtqYxHy:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_channel_secure.cc#L43)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

It looks like neither implementation (secure and insecure) of `ModifyGrpclbBalancerChannelArgs()` actually uses this argument.  Let's just remove it from the API and stop passing it in.
</blockquote></details>

Done.
___
*[src/core/lib/surface/server.cc, line 516 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0u-u34aF8j5xwBe0K:-MRSUtHH81nnn8Z3TS91:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/src/core/lib/surface/server.cc#L516)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

If this parameter isn't needed, let's just remove it from the API and change callers not to pass it.
</blockquote></details>

Done.
___
*[src/cpp/ext/filters/census/context.cc, line 33 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0siCwQSOFz3eRx3Tm-r1-33:-MRSV9fQ3Ijg9FJbv1WA:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/src/cpp/ext/filters/census/context.cc#L33)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

+1.  Doesn't seem very useful to have them here, since this isn't implementing some existing API.
</blockquote></details>

Done.
___
*[test/core/end2end/cq_verifier.cc, line 307 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0vWzz834oDrFIjFFh:-MRSWHaDDIjO87GD8Bkm:blnyyzu) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/test/core/end2end/cq_verifier.cc#L307)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

Hmm... Is this one a bug?  Are we sure that we shouldn't actually be using `success` here instead of hard-coding it to true?
</blockquote></details>

Oh yeah it looks a bit strange. Let's review this first. https://github.com/grpc/grpc/pull/19545 @yashykt Can you check this out?
___
*[test/cpp/end2end/xds_end2end_test.cc, line 1060 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0wYDu-ZQR8KKD8Ynv:-MRSWbci2-TAnP6ZKEQx:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/test/cpp/end2end/xds_end2end_test.cc#L1060)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

If this isn't actually being used, let's just remove the parameter.
</blockquote></details>

Done.

File: src/core/lib/transport/metadata.cc (hunk 1 above)
Comment (veblush) [reply src/core/lib/transport/metadata.cc:679]: This is needed because of `DEBUG_ARGS` which defines file and line. `DEBUG_ARGS` seems possible to remove for this function but i'm not 100% sure.

File: src/cpp/ext/filters/census/context.cc (hunk 2 above)
Comment (veblush) [reply src/cpp/ext/filters/census/context.cc]: Yeah I can remove this.

**Comment by veblush** [comment]
Rebased to address the merge conflict.

**Review by markdroth — COMMENTED** [review]
This looks great!  The only remaining issue is the one in cq_verifier.cc.

> Reviewed 13 of 14 files at r2.
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/25134#-:-MRVFEIe8vdS95ObXKyf:b-kv3him)* status: all files reviewed, 3 unresolved discussions (waiting on @drfloob, @gnossen, @nicolasnoble, and @yashykt)

**Review by yashykt — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/25134)* status: 48 of 60 files reviewed, 3 unresolved discussions (waiting on @drfloob, @gnossen, @markdroth, and @nicolasnoble)

___
*[test/core/end2end/cq_verifier.cc, line 307 at r1](https://reviewable.io/reviews/grpc/grpc/25134#-MR0vWzz834oDrFIjFFh:-MRWCtVS7X226OnnAcxK:b5qv19c) ([raw file](https://github.com/grpc/grpc/blob/fbd49f28fa9205450f01229fc5d7ac2c5e7c831d/test/core/end2end/cq_verifier.cc#L307)):*
<details><summary><i>Previously, veblush (Esun Kim) wrote…</i></summary><blockquote>

Oh yeah it looks a bit strange. Let's review this first. https://github.com/grpc/grpc/pull/19545 @yashykt Can you check this out?
</blockquote></details>

Yeah that does seem like a bug. It shouldn't be hardcoded.

**Review by markdroth — APPROVED** [review]
Looks great!  Thanks for doing this.

I'll go ahead and approve, and I'll let you and Yash figure out how to address my remaining comment.

> Reviewed 12 of 13 files at r4, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/25134#-:-MRWSkWS0sIyMXxrttu4:b-ootdom)* status: all files reviewed, 3 unresolved discussions (waiting on @drfloob, @gnossen, @nicolasnoble, and @veblush)

___
*[src/core/ext/xds/xds_api.cc, line 1392 at r5](https://reviewable.io/reviews/grpc/grpc/25134#-MRWSMtt9od6vJAeNXHz:-MRWSMtu3DHz4DdIutgK:b-sycosv) ([raw file](https://github.com/grpc/grpc/blob/985c8ff3b2ad159c9a22c64eedf651cfcd9c7a77/src/core/ext/xds/xds_api.cc#L1392)):*
> ```cpp
> grpc_error* LdsResponseParseServer(
>     upb_arena* arena, const envoy_config_listener_v3_Listener* listener,
>     const std::string& /*listener_name*/,
> ```

@yashykt Can we just remove these parameters if they're not needed?

**Review by yashykt — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/25134)* status: all files reviewed, 3 unresolved discussions (waiting on @drfloob, @gnossen, @markdroth, and @nicolasnoble)

___
*[src/core/ext/xds/xds_api.cc, line 1392 at r5](https://reviewable.io/reviews/grpc/grpc/25134#-MRWSMtt9od6vJAeNXHz:-MRa6yof5kR7eNO8HeYy:bivfcqa) ([raw file](https://github.com/grpc/grpc/blob/985c8ff3b2ad159c9a22c64eedf651cfcd9c7a77/src/core/ext/xds/xds_api.cc#L1392)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

@yashykt Can we just remove these parameters if they're not needed?
</blockquote></details>

I can fix it, I don't want to block this PE
